### PR TITLE
Add documentation for using carvel secret-gen-controller

### PR DESCRIPTION
**Pull request by dumez-k**

### CODE REVIEW

File: synced-secrets.html.md.erb [hunk 1]
```diff
@@ -18,13 +18,26 @@ Synchronized secrets are attached to build pods as `imagePullSecrets` so that th
...
+## <a id='install-secret-gen'></a> Installing the Carvel secret-gen-controller
+
+In order to manage secrets across namespaces, Tanzu Build Service utilizes the carvel secret-gen-controller. Tanzu Build Service will automatically create placeholder secrets in namespaces that contain Build resources and then the secret-gen-controller will populate these placeholders
+across namespaces with the corresponding sync-secret in the secret-syncer namespace.
+
+Please follow the secret-gen-controller [install docs](https://github.com/vmware-tanzu/carvel-secretgen-controller/blob/develop/docs/install.md) to install the controller.
+
+
+
+<p class='note'><strong>Note:</strong> As of TBS version 1.4 all secrets with the label "tbs-builder-secret-gen-placeholder-secret" will be deleted as this method of secret-syncing is deprecated. Please follow the steps below to update your secrets with the new label "com.vmware.tanzu.buildservice.placeholder"</p>
```
Comment (tylerphelan) [inline synced-secrets.html.md.erb:30]: old label is `com.vmware.tanzu.buildservice.sync`

Comment (tylerphelan) [reply synced-secrets.html.md.erb:30]: you also shouldn't need to set the new label: `com.vmware.tanzu.buildservice.placeholder`

File: synced-secrets.html.md.erb [hunk 2]
```diff
@@ -18,13 +18,26 @@ Synchronized secrets are attached to build pods as `imagePullSecrets` so that th
...
+
+
 ## <a id='manage-sync'></a> Managing Secret Synchronization
 
 Currently, the `kp` CLI does not support adding and removing synchronized secrets. However, this may be achieved by using the `kubectl` CLI.
 
 ### <a id='create-sync'></a> Create a Synchronized Secret
 
-To start synchronizing a secret to all namespaces with builds, use `kubectl` to create a docker-registry (Dockercfg or DockerConfigJson) secret in the `build-service` namespace with the following label: `com.vmware.tanzu.buildservice.sync=true`.
+To start synchronizing a secret to all namespaces with builds, use `kubectl` to create a docker-registry (Dockercfg or DockerConfigJson) secret in the `build-service` namespace with the following label: `com.vmware.tanzu.buildservice.placeholder=true`.
```
Comment (tylerphelan) [inline synced-secrets.html.md.erb:40]: I think the process is to create a SecretExport resource

Comment (matthewmcnew) [reply synced-secrets.html.md.erb:40]: +1

File: synced-secrets.html.md.erb (hunk 1 above)
Comment (matthewmcnew) [inline synced-secrets.html.md.erb:30]: Should this just be in the release notes?

**Comment by matthewmcnew** [comment]
Should this be updated to be "Using SecretGen controller Secrets to use private cluster builders?" (So, to reframe it as less of a TBS feature and more how we are making it possible to use SecretgenController?)

File: synced-secrets.html.md.erb [hunk 3]
```diff
@@ -18,13 +18,21 @@ Synchronized secrets are attached to build pods as `imagePullSecrets` so that th
...
+
+
 ## <a id='manage-sync'></a> Managing Secret Synchronization
 
 Currently, the `kp` CLI does not support adding and removing synchronized secrets. However, this may be achieved by using the `kubectl` CLI.
 
 ### <a id='create-sync'></a> Create a Synchronized Secret
 
-To start synchronizing a secret to all namespaces with builds, use `kubectl` to create a docker-registry (Dockercfg or DockerConfigJson) secret in the `build-service` namespace with the following label: `com.vmware.tanzu.buildservice.sync=true`.
+To start synchronizing a secret to all namespaces with builds, first create the secret in the secret-gen-controller namespace ("build-service") using the `kubectl` cli.
```
Comment (tylerphelan) [inline synced-secrets.html.md.erb:35]: maybe just `To start synchronizing a secret to all namespaces with builds, first create the secret in the `build-service` namespace using the `kubectl` cli.

File: synced-secrets.html.md.erb [hunk 4]
```diff
@@ -34,41 +42,34 @@ data:
...
+kind: SecretExport
 metadata:
-  labels:
-    com.vmware.tanzu.buildservice.sync: "true"
   name: my-synced-secret
   namespace: build-service
-type: kubernetes.io/dockerconfigjson
-EOF
+spec:
+  toNamespace: desired-export-namespace
```
Comment (tylerphelan) [inline synced-secrets.html.md.erb:61]: maybe do "*" here to do all build namespaces

File: synced-secrets.html.md.erb [hunk 5]
```diff
@@ -34,41 +42,34 @@ data:
...
+
 ### <a id='update-sync'></a> Update a Synchronized Secret
 
-To update a secret and roll-out those changes to all namespaces that use Builds, simply update the secret(s) with the `com.vmware.tanzu.buildservice.sync=true` label located in the `build-service` namespace.
+To update a secret and roll-out those changes to all namespaces that use Builds, simply update the secret(s) that have a corresponding secret-export resource located in the `build-service` namespace. Once this secret is updated the change will be rolled-out across namespaces.
 
 ### <a id='stop-sync'></a> Stop Synchronizing a Secret
 
-To stop synchronizing a secret, delete the secret from the `build-service` namespace or remove the `com.vmware.tanzu.buildservice.sync=true` label from the secret located in the `build-service` namespace.
+To stop synchronizing a secret, delete the secret from the `build-service` namespace or remove the corresonding secret-export resource from the secret located in the `build-service` namespace.
```
Comment (tylerphelan) [inline synced-secrets.html.md.erb:75]: `secret-export` -> `SecretExport` maybe

**Review by tylerphelan — CHANGES_REQUESTED** [review]
very nice

**Review by tylerphelan — APPROVED** [review]

**Pull request merged**